src/pages/wallet-overview/wallet-overview.ts

```diff
@@ -16,89 +19,187 @@ import { ICurrency } from '../../models/ICurrency';
 @Component({
   selector: "page-wallet-overview",
   templateUrl: "wallet-overview.html",
+  animations: [
+    trigger('visibilityChanged', [
+      state('shown', style({ opacity: 1 })),
+      state('hidden', style({ opacity: 0 , display: "none"})),
+      transition('* => *', animate('500ms'))
+    ])
+  ]
 })
 export class WalletOverviewPage {
   @ViewChild('doughnutCanvas') doughnutCanvas;
-  title = "Wallet Overview";
-  pickedCurrency = "$";
-  yellows: number;
-  blues: number;
-  currencyAmount: number;
-  smiloWorthPerCoin: number;
-  smiloPayWorthPerCoin: number;
+  pickedCurrency: string;
   doughnutChart: any;
   wallets: any = [];
   currenciesForDoughnutCanvas: any = [];
   currenciesForDoughnutCanvasCurrencies: any = [];
   currentWallet: any;
   currentWalletIndex = 0;
-  legendList: string[];
+  legendList: string[] = [];
   availableCurrencies: string[] = [];
+  showFundsStatus: boolean = true;
+  twoFactorStatus: boolean = false;
+  walletFundsVisibility: string = 'shown';
+  loading: Loading;
+  loadingError: Loading;
+
 
   constructor(public navCtrl: NavController, 
               public navParams: NavParams, 
-              public walletService: WalletService) {
-    this.currentWallet = "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ";
+              public walletService: WalletService,
+              public alertCtrl: AlertController,
+              public toastCtrl: ToastController,
+              public loadingCtrl: LoadingController) {
     this.getAllWallets();
     this.getAvailableCurrencies();
   }
 
   ionViewDidLoad() {
-    console.log("ionViewDidLoad WalletOverviewPage");
+    
+  }
+
+  fundsSwitch() {
+    if (this.walletFundsVisibility === "shown") {
+      this.walletFundsVisibility = "hidden";
+      this.showFundsStatus = false;
+    } else if (this.walletFundsVisibility === "hidden") {
+      this.walletFundsVisibility = "shown";
+      this.showFundsStatus = true;
+    }
+  }
+
+  twoFactorStatusSwitch() {
```

Comment:
You can simplify this by simply writing `this.twoFactorStatus = !this.twoFactorStatus`